compiler/rustc_driver/src/lib.rs

```diff
@@ -141,6 +141,9 @@ pub fn run_compiler(
     callbacks: &mut (dyn Callbacks + Send),
     file_loader: Option<Box<dyn FileLoader + Send + Sync>>,
     emitter: Option<Box<dyn Write + Send>>,
+    make_codegen_backend: Option<
```

Comment:
This argument list is slowly getting out of hand. Maybe we should create a `RunCompiler` builder pattern struct that doesn't require most calls to use `None, None, None`?

Anyway, pre-existing, if you rather open an issue about it than fix it here, I'm fine with merging this PR

Reply:
https://github.com/rust-lang/rust/issues/77286